src/main/java/edu/cornell/kfs/sys/dataaccess/impl/AutoCancelBatchDaoJdbc.java

```diff
@@ -0,0 +1,206 @@
+        final String daysToAutoCancel = parameterService.getParameterValueAsString(AutoCancelBatchStep.class, CUKFSParameterKeyConstants.DAYS_TO_AUTO_CANCEL_PARAMETER);
+
+        if (StringUtils.isNotBlank(daysToAutoCancel)) {
+            Map<String, String> cancelIds = findSavedDocumentIds(Integer.parseInt(daysToAutoCancel));
+            Set<String> cancelDocumentIds = cancelIds.keySet();
+
+            int canceledDocumentCount = 0;
+            for (String docId : cancelDocumentIds) {
+                String docTypeId = cancelIds.get(docId);
+                if (canAutoCancelDocType(docTypeId)) {
+                    LOG.info("Retrieving document : " + docId.trim());
+                    Document document = documentService.getByDocumentHeaderId(docId.trim());
+
+                    try {
+                        if (!ObjectUtils.isNull(document)) {
+                            LOG.info("Document Number to cancel : " + document.getDocumentNumber());
+                            canceledDocumentCount++;
+                            documentService.prepareWorkflowDocument(document);
+                            workflowDocumentService.superUserCancel(document.getDocumentHeader().getWorkflowDocument(), "AutoCancelBatchStep: Older Than " + daysToAutoCancel + " Days");
+                            sessionDocumentService.addDocumentToUserSession(GlobalVariables.getUserSession(), document.getDocumentHeader().getWorkflowDocument());
+                        }
+                    } catch (WorkflowException e) {
+                        LOG.error("AutoCancelBatchStep Encountered WorkflowException " + document.getDocumentNumber(), e);
+                    }
+                }
+            }
+            LOG.info("Total number of docs canceled : " + canceledDocumentCount);
+        } else {
+            LOG.info("ERROR: DAYS_TO_CANCEL parameter is empty or missing");
+        }
+
+    }
+
+    private Map<String, String> findSavedDocumentIds(int daysToAutoCancel) {
+        Map<String, String> ids = new HashMap<String, String>();
+
+        StringBuilder sql = new StringBuilder();
+        sql.append("select DOC_HDR_ID, DOC_TYP_ID from CYNERGY.KREW_DOC_HDR_T where ");
+        sql.append("DOC_HDR_STAT_CD = '" + DocumentStatus.SAVED.getCode() + "' and (trunc(CRTE_DT) + " + daysToAutoCancel + ") <= trunc(SYSDATE)");
+
+        LOG.info("SQL Statement: " + sql);
+        List<Map<String, Object>> results = getSimpleJdbcTemplate().queryForList(sql.toString());
+
+        for (Map<String, Object> result : results) {
+            String docId = (String) result.get("DOC_HDR_ID");
+            String docTypId = (String) result.get("DOC_TYP_ID");
+            LOG.info("Doc ID : " + docId + " Doc Typ ID : " + docTypId);
+            ids.put(docId, docTypId);
+        }
+
+        return ids;
+    }
+
+    private boolean canAutoCancelDocType(String docTypeId) {
+        Map<String, DocumentType> docTypes = new HashMap<String, DocumentType>();

```

Comment:
This helper Map only exists locally in this method, but the code below is acting as though the Map is maintaining its state between method calls. If there is a need to temporarily cache these document types, then a new field should be added to this class accordingly.